src/services/codeAssistant.ts

```diff
@@ -127,27 +144,167 @@ export default class CodeAssistantService extends ServiceAPI {
         stream: streamingEnabled
       })
     };
-  
+
     if (streamingEnabled) {
-      const response = ServiceAPI.runFetchStreaming(endpoint, options);
-  
-      for await (let chunk of response) {
-        // parse & transform the streaming data chunk
-        const lines = chunk.split('\n');
-        for (let i = 0; i < lines.length; i++) {
-          const line = lines[i].trim();
-          if (line.startsWith(STREAM_DATA_PREFIX)) {
-            try {
-              // remove 'data: ' prefix and parse remaining string
-              const jsonChunk = JSON.parse(line.substring(STREAM_DATA_PREFIX.length)) as ModelPromptResponse;
-              yield jsonChunk;
-            } catch (error) {
-              // JSON parsing errors
-              console.error(`Error parsing JSON: ${error}`);
-              console.log(line)
+      // Get circuit breaker instance
+      const circuitBreaker = circuitBreakerEnabled
+        ? getCircuitBreaker(SERVICE_NAME, {
+            failureThreshold: circuitBreakerThreshold,
+            resetTimeout: circuitBreakerResetTimeout,
+            successThreshold: 2
+          })
+        : null;
+
+      try {
+        // Check circuit breaker before request
+        if (circuitBreaker) {
+          await circuitBreaker.beforeRequest();
+        }
+
+        // Log streaming start
+        if (enableMetrics && retryCount === 0) {
+          StreamingTelemetry.logStart('postModelPrompt', {
+            modelId,
+            service: SERVICE_NAME,
+            streamingEnabled: true,
+            bufferSize,
+            timeout
+          });
+        }
+
+        // Start streaming status indicator
+        if (retryCount === 0) {
+          streamingStatusBar.start("Generating completion");
+        } else {
+          streamingStatusBar.start(`Retrying... (${retryCount}/${maxRetries})`);
+        }
```

Comment:
curious, are these string messages referenced anywhere? i did not see them being used in the `StreamingStatusBarManager`